src/base/bittorrent/torrenthandle.cpp

```diff
@@ -527,11 +527,19 @@ qreal TorrentHandle::progress() const
     if (m_nativeStatus.total_wanted_done == m_nativeStatus.total_wanted)
         return 1.;
 
-    float progress = static_cast<float>(m_nativeStatus.total_wanted_done) / m_nativeStatus.total_wanted;
+    qreal progress = static_cast<qreal>(m_nativeStatus.total_wanted_done) / m_nativeStatus.total_wanted;
     Q_ASSERT((progress >= 0.f) && (progress <= 1.f));
     return progress;
 }
 
+qreal TorrentHandle::taskProgress() const
```

Comment:
Unfortunately, now I have noticed that the current `progress()` method does not follow the download progress when the "check" occurs, but actually returns the checking progress. So this additional method turns out to be useless.

Reply:
>the current progress() method does not follow the download progress when the "check" occurs

In my experience, the progress() will return the same value as before the recheck until the recheck finishes and updates which pieces we have. I observed this for torrents that I manually do a recheck.
Broken torrents probably won't return something in progress() until the recheck has finished.

In any case, should I drop this new method and incorporate it again into progress()?